### PR TITLE
cli: error early if --output-path is invalid

### LINKED ISSUE #14361: Improve error response time
**Feature request summary**
It would be better to check if the output destination is accessible first when the --output-path option is used during the lightHouse run.
The current behavior is that if the --output-path option is used incorrectly, an error is output after all work.

**What is the motivation or use case for changing this?**
Because it was not a good experience for me.

**How is this beneficial to Lighthouse?**
* Improved usability
* Prevent unnecessary resource use

### PR DESCRIPTION
**Pull request by connorjclark**
fixes #14361

### CODE REVIEW

**Review by adamraine — APPROVED** [review]

**Comment by brendankenny** [comment]
I feel like there's a super-ancient TODO about this that can maybe finally be deleted, but maybe it was already deleted...

File: cli/cli-flags.js [hunk 1]
```diff
@@ -412,13 +412,13 @@ function coerceOutput(values) {
...
  * Verifies outputPath is something we can actually write to.
- * @param {unknown} value
+ * @param {unknown=} value
  * @return {string=}
  */
 function coerceOutputPath(value) {
   if (value === undefined) return;
 
-  if (typeof value !== 'string' || !fs.existsSync(path.dirname(value))) {
+  if (typeof value !== 'string' || !value || !fs.existsSync(path.dirname(value))) {
```
Comment (adamraine) [inline cli/cli-flags.js:421]: In what situation will `value === undefined`.

If it's the case where `--output-path` is provided without any value, could we do a different error message for that?

**Comment by brendankenny** [comment]
> I feel like there's a super-ancient TODO about this that can maybe finally be deleted, but maybe it was already deleted...

oh, ha, still no:
https://github.com/GoogleChrome/lighthouse/blob/9acca3a60c7c8c4d959e1e36fc021eddce488ced/cli/printer.js#L61

humorously we'll create[ the path for `-G` in `saveArtifacts`](https://github.com/GoogleChrome/lighthouse/blob/9acca3a60c7c8c4d959e1e36fc021eddce488ced/core/lib/asset-saver.js#L112), but still not for a path for just saving the LHR.

File: cli/cli-flags.js (hunk 1 above)
Comment (connorjclark) [reply cli/cli-flags.js:421]: > In what situation will value === undefined.

when not provided. coerce is still called when param is not specified (or at least, it is for the "flags in a json path" case).

> If it's the case where --output-path is provided without any value, could we do a different error message for that?

I think the current error is sufficient.

**Pull request merged**